application/forms/EventRuleConfigForm.php

```diff
@@ -309,14 +309,18 @@ public function storeInDatabase(Connection $db, ?Rule $previousRule): int
         }
 
         $escalations = $this->getElement('escalations')->getEscalations();
-        $currentEscalationIDs = array_map(
-            function ($escalation) {
-                return $escalation->getEscalation()['id'];
-            },
-            $escalations
-        );
+        $remainingDbEscalations = [];
+        $escalationConfigs = [];
+        foreach ($escalations as $escalation) {
+            $config = $escalation->getEscalation();
+            $escalationConfigs[] = $config;
```

Comment:
it's just slight safe-guard, but since you address the entries in `$escalationConfigs` by the position of an escalation in `$escalations` below, you should also use the same position when inserting them here.